### DISCUSSION TITLE
API to get the progress bar

### DISCUSSION THREAD

**Discussion by jamalsa** [duckdb/duckdb#9066] [Ideas]
I am developing a GUI application using DuckDB as its engine. For long query DuckDB will display progress bar on terminal. I want to display this progress bar in the UI. Is there any API to get the progress bar or progress percentage?

**Comment by Tishj** [comment]
It's possible to provide a custom progress bar to the config in the c++ API, you should look into that

**Reply by jamalsa** [reply]
Thanks @Tishj , but I am using Python client. Can I provide custom progress bar using Python?

**Reply by Tishj** [reply]
Not out of the box no

But for running in Jupyter we also use a different progress bar, so there is precedence and shouldn't be too difficult to add generically

**Reply by Mause** [reply]
Something similar to the sqlite3 based one maybe? https://docs.python.org/3/library/sqlite3.html?highlight=sqlite3#sqlite3.Connection.set_progress_handler

**Reply by Tishj** [reply]
Yea I have a branch that works on the `set_progress_handler` method, that's a little more fine-grained than the custom progress bar
Because the `set_progress_handler` just registers a callback that gets triggered every time the progress bar is updated

**Reply by Tishj** [reply]
@Mause <https://github.com/Tishj/duckdb/pull/26> I left it because some issues popped up that didn't seem trivial and demand for the feature is (was?) low

**Reply by jamalsa** [reply]
@Tishj this looks good. I will do some test with your fork. Can you tell me what issues popped up before?

**Reply by Tishj** [reply]
> @Tishj this looks good. I will do some test with your fork. Can you tell me what issues popped up before?

There is a comment in he linked PR that explains the issue(s)
